File: leetcode/1679_max_number_of_k_sum_pairs/solution.cpp
class Solution {
public:
    int maxOperations(vector<int>& nums, int k) {
        unordered_map<int, int> freq;
        int result = 0;

        for (int num : nums) {

            if (freq[k - num] > 0) {
                result++;
                freq[k - num]--;

            } else {
                freq[num]++;
            }
        }

        return result;
    }
};

File: leetcode/1679_max_number_of_k_sum_pairs/solution.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        nums.sort()
        left, right = 0, len(nums) - 1
        result = 0

        while left < right:
            s = nums[left] + nums[right]
            if s == k:
                result += 1
                left += 1
                right -= 1
            elif s < k:
                left += 1
            else:
                right -= 1

        return result
